### ISSUE TITLE
How training batch is generated when resuming training from checkpoint

### ISSUE THREAD

**Issue by jqJordan** [facebookresearch/detectron2#829]
If I resume training from a checkpoint for mask rcnn, how is training batch generated. Suppose I have 1000 images, and save a checkpoint after 100 iteration, that is, 200 images has been trained on(mini batch size 2),then reload the data and checkpoint, and rerun training,  will the batch be sampled from the rest 800 images or will the sampling start with the entire dataset again? Thank you. Because with limited resource, I have to train the model for several times, and I just want to make sure even I resume training from previous checkpoints, all images will be trained on.

**Comment by ppwwyyxx** [comment]
Training data are randomly sampled.
